controllers/secret/secret_controller.go

```diff
@@ -213,6 +223,28 @@ func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result
 	}
 }
 
+// removeDepreciatedAnnotations corrects annotations applied by previous versions of WMCO.
+func (r *ReconcileSecret) removeDepreciatedAnnotations() error {
+	nodes := &core.NodeList{}
+	err := r.client.List(context.TODO(), nodes)
+	if err != nil {
+		return errors.Wrapf(err, "error getting node list")
+	}
+	escapedPubKeyAnnotation := strings.Replace(nodeconfig.PubKeyHashAnnotation, "/", "~1", -1)
+	patchData := fmt.Sprintf(`[{"op":"remove","path":"/metadata/annotations/%s"}]`, escapedPubKeyAnnotation)
+	for _, node := range nodes.Items {
+		// The public key hash was accidentally added to Linux nodes in WMCO 2.0, this needs to be removed
+		if _, present := node.Annotations[nodeconfig.PubKeyHashAnnotation]; present == true &&
+			node.Labels[core.LabelOSStable] != "windows" {
```

Comment:
If you only list Linux nodes using the label selector you won't have to do this check.